src/layers/TrafimageMapboxLayer/TrafimageMapboxLayer.js

```diff
@@ -66,15 +67,29 @@ class TrafimageMapboxLayer extends MapboxLayer {
     });
   }
 
+  setStyle(newStyle) {
+    this.style = newStyle || this.options.style;
+    this.setStyleConfig();
+  }
+
   setStyleConfig(url, key, apiKeyName) {
-    if (!url) {
-      return;
+    if (!url && !key && !apiKeyName) {
+      this.styleUrl = `${this.url}/styles/${this.style}/style.json?`;
+    }
+    if (url && url !== this.url) {
+      this.url = url;
+    }
+    if (key && key !== this.apiKey) {
+      this.apiKey = key;
+    }
+    if (apiKeyName && apiKeyName !== this.apiKeyName) {
+      this.apiKeyName = apiKeyName;
     }
-    const { style } = this.options;
-    const newStyleUrl = `${url}/styles/${style}/style.json${
-      key ? `?${apiKeyName || 'key'}=${key}` : ''
-    }`;
 
+    const newStyleUrl =
+      this.styleUrl && this.style
+        ? this.createStyleUrl()
+        : `${this.url}/styles/${this.style}/style.json?`;
```

Comment:
Should  it be 
this.styleUrl = this.createStyleUrl(this.apiKey, this.apiKeyName , this.styleUrl); 
?